dev-proxy/ProxyEngine.cs

```diff
@@ -95,19 +99,14 @@ public async Task Run(CancellationToken? cancellationToken)
 
         var ipAddress = string.IsNullOrEmpty(_config.IPAddress) ? IPAddress.Any : IPAddress.Parse(_config.IPAddress);
         _explicitEndPoint = new ExplicitProxyEndPoint(ipAddress, _config.Port, true);
-        if (!RunTime.IsWindows)
+        // Fired when a CONNECT request is received
+        _explicitEndPoint.BeforeTunnelConnectRequest += OnBeforeTunnelConnectRequest;
+        _explicitEndPoint.GenericCertificate = _proxyServer.CertificateManager.LoadRootCertificate();
+        if (_config.InstallCert)
```

Comment:
Shouldn't we keep `!RunTime.IsWindows` here to avoid registering the cert twice? Since it's already being done by Windows as part of `SetAsSystemProxy` on line 124, we only need this for non-Windows OSes

Reply:
It would only be added once but sure, doesn't hurt.

Reply:
Now i know, why i did that: `devproxy --as-system-proxy false`
If you run that on windows, it will not install the cert even though it should be installed because the default for `--install-cert` is `true`.